### PR TITLE
Hive metastore connection pool optimization, Fixes for: backward compatibility for Hive in AvroToOrc, schema parser deserialization from schema literal, database name in Hive DDL query generation, Hive metastore connection pool initialization NPE if Hcat uri is platform provided

### PR DESCRIPTION
**Pull request by abti**
- Fix for backward compatibility for Hive in AvroToOrc
- Fix for schema parser deserialization from schema literal
- Fix for database name in Hive DDL query generation
- Fix for Hive metastore connection pool initialization NPE if Hcat uri is platform provided

### CODE REVIEW

**Comment by abti** [comment]
@pcadabam Please review.

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java [hunk 1]
```diff
@@ -55,6 +57,7 @@
  * If multiple {@link Partition}s have the same {@link Schema} a duplicate schema file in not created. Already existing
  * {@link Schema} url for this {@link Schema} is used.
  */
+@Slf4j
 public class AvroSchemaManager {
 
   private static final String HIVE_SCHEMA_TEMP_DIR_PATH_KEY = "hive.schema.dir";
```
Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java]: Can you delete the instance Schema.Parser instance here?

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java [hunk 2]
```diff
@@ -115,16 +118,22 @@ public static Schema getSchemaFromUrl(Path schemaUrl, FileSystem fs) throws IOEx
...
-      Schema schema =
-          this.schemaParser.parse(sd.getSerdeInfo().getParameters().get(HiveAvroSerDeManager.SCHEMA_LITERAL));
-      return getOrGenerateSchemaFile(schema);
-
-    } else {
-      Schema schema = AvroUtils.getDirectorySchema(new Path(sd.getLocation()), this.fs, true);
-      return getOrGenerateSchemaFile(schema);
     }
+    if (sd.getSerdeInfo().getParameters().containsKey(HiveAvroSerDeManager.SCHEMA_LITERAL)) {
+      String schemaLiteral = null;
```
Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java]: Move inside the try

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java (hunk 1 above)
Comment (abti) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/AvroSchemaManager.java]: Done

**Pull request merged**